src/transformStateWithClones.js

```diff
@@ -1,13 +1,41 @@
 'use strict';
 
 /**
- * @param {Object} state
+ * @param {Object} stateCopy
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const STATE_COPYS = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    let stateCopy = { ...currentState };
+
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const keyRemove of action.keysToRemove) {
+          if (stateCopy[keyRemove] !== undefined) {
+            delete stateCopy[keyRemove];
+          }
+        }
+        break;
+
+      case 'clear':
+        stateCopy = {};
+        break;
+    }
```

Comment:
Checklist:
- [CODE STYLE]: switch/case should always have default case for error handling.